Numsense/German.fs

```diff
@@ -0,0 +1,159 @@
+module internal Ploeh.Numsense.German
+
+open Ploeh.Numsense.InternalDsl
+open System
+
+let rec convertPart einEnde x  =
+
+    let simplifyReverse prefix factor x =
+        let remainder = x % factor
+        match remainder = 0 with
+        | true ->
+            prefix
+        | false ->
+            sprintf "%s-und-%s" (convertPart "" (remainder) ) prefix
+
+    let simplify prefix factor x =
+        let remainder = x % factor
+        match remainder = 0 with
+        | true ->
+            prefix
+        | false ->
+            sprintf "%s-%s" prefix (convertPart "" (remainder))
+
+    let format suffix factor x =
+        let prefix = sprintf "%s%s" (convertPart "" (x / factor)) suffix
+        simplify prefix factor x
+
+    match x with
+    |  1 -> "ein" + einEnde
+    |  2 -> "zwei"
+    |  3 -> "drei"
+    |  4 -> "vier"
+    |  5 -> "fünf"
+    |  6 -> "sechs"
+    |  7 -> "sieben"
+    |  8 -> "acht"
+    |  9 -> "neun"
+    | 10 -> "zehn"
+    | 11 -> "elf"
+    | 12 -> "zwölf"
+    | 13 -> "dreizehn"
+    | 14 -> "vierzehn"
+    | 15 -> "fünfzehn"
+    | 16 -> "sechzehn"
+    | 17 -> "siebzehn"
+    | 18 -> "achtzehn"
+    | 19 -> "neunzehn"
+    | 20 -> "zwanzig"
+    | Between 20 30 x -> simplifyReverse "zwanzig" 10 x
+    | Between 30 40 x -> simplifyReverse "dreißig" 10 x
+    | Between 40 50 x -> simplifyReverse "vierzig" 10 x
+    | Between 50 60 x -> simplifyReverse "fünfzig" 10 x
+    | Between 60 70 x -> simplifyReverse "sechzig" 10 x
+    | Between 70 80 x -> simplifyReverse "siebzig" 10 x
+    | Between 80 90 x -> simplifyReverse "achtzig" 10 x
+    | Between 90 100 x -> simplifyReverse "neunzig" 10 x
+    | 100 -> format "-hundert" 100 x
+    | 101 -> "ein-hundert-eins"
+    | Between 101 1000 x -> format "-hundert" 100 x
+    | _ -> ""
+
+let convToInt (str : string) =
+    Convert.ToInt32(str)
+
+let combineParts (newPart : string) (result : string) =
+    match (newPart <> "") with
+    | true ->
+        match (result <> "") with
+        | true -> sprintf "%s-%s" newPart result
+        | false -> newPart
+    | false -> result
+
+let internal toGermanNumber (x : int ) =
+
+    let minus = (x < 0)
+    let stringValue = Math.Abs(x).ToString().PadLeft(10, '0')
+
+    let milliarde = stringValue.Substring(0,1) |> convToInt |> convertPart ""
+    let million = stringValue.Substring(1,3) |> convToInt |> convertPart ""
+    let tausend = stringValue.Substring(4,3) |> convToInt |> convertPart ""
+    let rest = stringValue.Substring(7,3) |> convToInt |> convertPart "s"
```

Comment:
It's an interesting approach to the problem.
Just a question, couldn't it be achieved by dividing the number by the corresponding factor?
I mean without transforming it to a string and splitting it on positions, it looks little hacky IMHO.